compiler/hash-ast/src/lit.rs

```diff
@@ -21,8 +21,8 @@ use std::num;
 use hash_reporting::{hash_error_codes::error_codes::HashErrorCode, reporter::Reporter};
```

Comment:
Do you mean the parsing part or the actual representation?

Reply:
The whole file essentially

Reply:
Where could it be instead, in the parser?

Reply:
Since literals generally hang around hash-source, maybe hash-source?

Reply:
Perhaps this should be in `hash-source`

Reply:
On second thought, it can't really be in `hash-source` because it defines several things (such as the error stuff) which depends on crates like `hash-reporting`. I think it is sensible to leave it in `hash-ast` or perhaps move it somewhere else that can be dependent on `hash-token`, `hash-ast`, `hash-source`, `hash-reporting`. The only sensible other alternative to `hash-ast` is perhaps `hash-parser`.

Reply:
I see... Do you think it makes sense to have a crate specifically for literals? It's just that these structures do not appear anywhere in the AST so it doesn't really make sense for them to be here..